capstone/src/test/java/com/google/sps/servlets/announcements/EditAnnouncementServletTest.java

```diff
@@ -0,0 +1,154 @@
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void doPost_authorizedEdit() throws ServletException, IOException {
+    prepare();
+    helper.setEnvEmail(TEST_EMAIL).setEnvAuthDomain("google.com").setEnvIsLoggedIn(true);
+
+    String id = TEST_EMAIL + SAMPLE_CONTENT + SAMPLE_TIME;
+    when(request.getParameter(Constants.CONTENT_PROP)).thenReturn(SAMPLE_NEW_CONTENT);
+    when(request.getParameter(Constants.CLUB_PROP)).thenReturn(SAMPLE_CLUB_NAME);
+    when(request.getParameter(ID_PROP)).thenReturn(id);
+    servlet.doPost(request, response);
+
+    Query query = new Query(Constants.ANNOUNCEMENT_PROP);
+    PreparedQuery results = datastore.prepare(query);
+
+    String newId = TEST_EMAIL + SAMPLE_NEW_CONTENT + SAMPLE_TIME;
+    ImmutableList<Entity> entities =
+        Streams.stream(results.asIterable())
+            .filter(
+                entity ->
+                    SAMPLE_CLUB_NAME.equals(entity.getProperty(Constants.CLUB_PROP))
+                        && newId.equals(
+                            entity.getProperty(Constants.AUTHOR_PROP).toString()
+                                + entity.getProperty(Constants.CONTENT_PROP).toString()
+                                + entity.getProperty(Constants.TIME_PROP).toString()))
+            .collect(toImmutableList());
+
+    Entity entity = entities.get(0);
+
+    Assert.assertNotNull(entity);
+    Assert.assertEquals(SAMPLE_NEW_CONTENT, entity.getProperty(Constants.CONTENT_PROP));
+  }
+
+  @Test
+  public void doPost_unauthorizedEdit() throws ServletException, IOException {
+    prepare();
+    helper
+        .setEnvEmail("anotherEmail@google.com")
+        .setEnvAuthDomain("google.com")
+        .setEnvIsLoggedIn(true);
+
+    String id = TEST_EMAIL + SAMPLE_CONTENT + SAMPLE_TIME;
+    when(request.getParameter(Constants.CONTENT_PROP)).thenReturn(SAMPLE_NEW_CONTENT);
+    when(request.getParameter(Constants.CLUB_PROP)).thenReturn(SAMPLE_CLUB_NAME);
+    when(request.getParameter(ID_PROP)).thenReturn(id);
+    servlet.doPost(request, response);
+
+    Query query = new Query(Constants.ANNOUNCEMENT_PROP);
+    PreparedQuery results = datastore.prepare(query);
+
+    ImmutableList<Entity> entities =
+        Streams.stream(results.asIterable())
+            .filter(
+                entity ->
+                    SAMPLE_CLUB_NAME.equals(entity.getProperty(Constants.CLUB_PROP))
+                        && id.equals(
+                            entity.getProperty(Constants.AUTHOR_PROP).toString()
+                                + entity.getProperty(Constants.CONTENT_PROP).toString()
+                                + entity.getProperty(Constants.TIME_PROP).toString()))
+            .collect(toImmutableList());
+
+    Entity entity = entities.get(0);
+
+    Assert.assertNotNull(entity);
+    Assert.assertEquals(SAMPLE_CONTENT, entity.getProperty(Constants.CONTENT_PROP));
+  }
+
+  private void prepare() {
```

Comment:
Since this is needed before all tests in this file, would it work to be it in the setup?